library/src/main/java/com/google/android/exoplayer2/text/eia608/Eia608Decoder.java

```diff
@@ -414,32 +501,201 @@ private void handleMiscCode(byte cc2) {
+      // Foreground color
+      captionStyles.put(getSpanStartIndex(), new ForegroundColorSpan(COLOR_MAP[attribute]));
+      if (transparentOrUnderline) {
+        // Text should be underlined
+        captionStyles.put(getSpanStartIndex(), new UnderlineSpan());
+      }
+    }
+  }
+
+  private void handleTabOffset(byte cc2) {
+    // Formula for tab offset handling adapted from:
+    // https://dvcs.w3.org/hg/text-tracks/raw-file/default/608toVTT/608toVTT.html#x1-preamble-address-code-pac
+    // We're ignoring any tab offsets that do not occur at the beginning of a new cue.
+    // This is not conform the spec, but works in most cases.
+    if (captionStringBuilder.length() == 0) {
+      switch (cc2) {
+        case CTRL_TAB_OFFSET_1:
+          tabOffset++;
+          break;
+        case CTRL_TAB_OFFSET_2:
+          tabOffset += 2;
+          break;
+        case CTRL_TAB_OFFSET_3:
+          tabOffset += 3;
+          break;
+      }
+    }
+  }
+
+  private int getSpanStartIndex() {
+    return captionStringBuilder.length() > 0 ? captionStringBuilder.length() - 1 : 0;
+  }
+
+  /**
+   * Applies a Span to the SpannableStringBuilder.
+   */
+  private void applySpan() {
+    // Check if we have to do anything.
+    if (captionStyles.size() == 0) {
+      return;
+    }
+
+    for (Integer startIndex : captionStyles.keySet()) {
+      // There may be cases, e.g. when seeking where the startIndex becomes greater
+      // than what is actually in the string builder, in that case, just discard the span.
+      if (startIndex < captionStringBuilder.length()) {
+        CharacterStyle captionStyle = captionStyles.get(startIndex);
+        captionStringBuilder.setSpan(captionStyle, startIndex,
+          captionStringBuilder.length(), Spanned.SPAN_EXCLUSIVE_EXCLUSIVE);
+      }
+      captionStyles.remove(startIndex);
+    }
+  }
+
+  /**
+   * Builds a cue from whatever is in the SpannableStringBuilder now.
+   */
+  private void buildCue() {
+    applySpan(); // Apply Spans
+    CharSequence captionString = getDisplayCaption();
+    if (captionString != null) {
+      cueIndent = tabOffset * 2.5f + cueIndent;
+      tabOffset = 0;
+      Cue cue = new Cue(captionString, Layout.Alignment.ALIGN_NORMAL, cueLine / 100, Cue.LINE_TYPE_FRACTION,
+        Cue.ANCHOR_TYPE_START, cueIndent / 100, Cue.TYPE_UNSET, Cue.DIMEN_UNSET);
+      cues.add(cue);
+      if (captionMode == CC_MODE_POP_ON) {
+        captionStringBuilder.clear();
+        captionStringBuilder.clearSpans();
+        cueLine = DEFAULT_CUE_LINE;
+      }
+      cueIndent = DEFAULT_INDENT;
+    }
+  }
+
+  private void flushCaptionBuilder() {
+    captionStringBuilder.clear();
+    captionStringBuilder.clearSpans();
+  }
+
   private void backspace() {
     if (captionStringBuilder.length() > 0) {
-      captionStringBuilder.setLength(captionStringBuilder.length() - 1);
+      captionStringBuilder.replace(captionStringBuilder.length() - 1, captionStringBuilder.length(), "");
```

Comment:
According to the [documentation](https://developer.android.com/reference/android/text/SpannableStringBuilder.html#delete(int, int)) delete is just a convenience method, but I agree, it would probably fit better here.